node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/History.java

```diff
@@ -63,7 +63,8 @@ public History without(Event.Type type) {
 
     /** Returns a copy of this history with a record of this state transition added, if applicable */
     public History recordStateTransition(Node.State from, Node.State to, Agent agent, Instant at) {
-        if (from == to) return this;
+        // If the event is a re-reservation, allow the new one to override the older one.
```

Comment:
How about just removing the shortcut?

Reply:
I assumed it was there to prevent accidental overrides of data, if a no-op transition was attempted. You have the context here, so feel free to take this call :)

Reply:
Yes, you may be right ... at least I don't want to think about it now.